File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/editor/codestyle/java/javadoc/JavadocInvalidTagsSettings.java
package com.github.alexandrecarlton.idea.settings.layout.editor.codestyle.java.javadoc;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavadocInvalidTagsSettings.class)
public interface JavadocInvalidTagsSettings {

  Optional<Boolean> keepInvalidTags();

  Optional<Boolean> keepEmptyParamTags();

  Optional<Boolean> keepEmptyReturnTags();

  Optional<Boolean> keepEmptyThrowsTags();
}

File: layout/src/main/kotlin/com/github/alexandrecarlton/idea/settings/dialog/editor/codestyle/java/javadoc/JavadocInvalidTagsSettings.kt
package com.github.alexandrecarlton.idea.settings.dialog.editor.codestyle.java.javadoc

import com.fasterxml.jackson.annotation.JsonProperty

data class JavadocInvalidTagsSettings(

    @JsonProperty("Keep invalid tags")
    val keepInvalidTags: Boolean? = null,

    @JsonProperty("Keep empty @param tags")
    val keepEmptyParamTags: Boolean? = null,

    @JsonProperty("Keep empty @return tags")
    val keepEmptyReturnTags: Boolean? = null,

    @JsonProperty("Keep empty @throws tags")
    val keepEmptyThrowsTags: Boolean? = null
)
